interceptors/timeout/timeout.go

```diff
@@ -0,0 +1,17 @@
+package timeout
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+)
+
+// TimeoutUnaryClientInterceptor returns a new unary client interceptor that sets a timeout on the request context.
+func TimeoutUnaryClientInterceptor(timeout time.Duration) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		timedCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
```

Comment:
Just curious, how this is different to setting timeout on gRPC client side, or passing context with timeout?

Reply:
this middleware just ensure that all grpc call has a "default timeout".
in other hand i have commit c747443231c71873c05b7b9ec2d512e14cf45b9c a fix where the incoming context already has timeout (it will keep it)

Reply:
Cool: https://github.com/grpc/grpc-go/issues/2130